Data/ByteString/Lazy.hs

```diff
@@ -696,6 +726,52 @@ drop i cs0 = drop' i cs0
             then Chunk (S.drop (fromIntegral n) c) cs
             else drop' (n - fromIntegral (S.length c)) cs
 
+-- | /O(n)/ @'dropEnd' n xs@ is equivalent to @'take' ('length' xs - n) xs@.
+-- Drops @n@ elements from end of bytestring.
+--
+-- >>> dropEnd 3 "abcdefg"
+-- "abcd"
+-- >>> dropEnd 0 "abcdefg"
+-- "abcdefg"
+-- >>> dropEnd 4 "abc"
+-- ""
+--
+-- @since 0.11.2.0
+dropEnd :: Int64 -> ByteString -> ByteString
+dropEnd i p | i <= 0 = p
+dropEnd i p = go D.empty p
+  where go :: D.Deque -> ByteString -> ByteString
+        go deque (Chunk c cs)
+            | D.elemLength deque < i = go (D.snoc c deque) cs
+            | otherwise              =
+                  let (output, deque') = getOutput [] (D.snoc c deque)
+                    in L.foldl (flip chunk) (go deque' cs) output
+        go deque Empty               = dropElements deque (fromIntegral i)
+
+        len c = fromIntegral (S.length c)
+
+        -- get all `S.ByteString` from the front of the accumulating deque
+        -- for which we know they won't be dropped
+        getOutput :: [S.ByteString] -> D.Deque -> ([S.ByteString], D.Deque)
+        getOutput out deque = case D.popFront deque of
+            Nothing                       -> (out, deque)
+            Just (x, deque') | D.elemLength deque' >= i -> getOutput (x:out) deque'
+                             | otherwise  -> (out, deque)
+
+        -- drop n elements from the rear of the accumulating `deque`
```

Comment:
The term "elements" is slightly confusing in this context: Does it mean the bytes or the individual strict bytestring chunks contained in the deque? I'd suggest using the term "bytes".

Reply:
I used elements as its consistently used to describe bytes in the module (eg. documentation for `drop`, `takeWhile` etc). The same applies for strict bytestirngs (e.g. S.dropEnd  also uses 'elements'). Is this reasoning sufficient?

Reply:
In this case you seem to refer to the elements of the deque. However the operations implemented for the `Deque` type all seem to handle strict bytestrings as elements. So I think that "bytes" would be clearer. Maybe you can find a different wording that clarifies things though.